client/core/core.go

```diff
@@ -3492,6 +3510,39 @@ func handlePreimageRequest(c *Core, dc *dexConnection, msg *msgjson.Message) err
 	var oid order.OrderID
 	copy(oid[:], req.OrderID)
 
+	// Sync with order placement.
+	c.piSyncMtx.Lock()
+	syncChan, found := c.piSyncers[oid]
+	if found {
+		// If we found the channel, the tracker is already in the map, and we
+		// can go ahead.
+		delete(c.piSyncers, oid)
+	} else {
+		// If we didn't find the channel, Trade is still running. Add the chan
+		// and wait for Trade to close it.
+		syncChan = make(chan struct{})
+		c.piSyncers[oid] = syncChan
+	}
+	c.piSyncMtx.Unlock()
+
+	deletePiSyncer := func() {
+		c.piSyncMtx.Lock()
+		delete(c.piSyncers, oid)
+		c.piSyncMtx.Unlock()
+	}
+
+	if !found {
+		select {
+		case <-syncChan:
+		case <-time.After(time.Minute):
```

Comment:
Ah, we've neglected cancel orders, which are not generated in `prepareTrackedTrade`, but `tryCancel`.

Also, while you are in `(*dexConnection).tryCancel` or `(*Core).Cancel`, it would be nice to log the submission of the cancel order and what it's targeting and the oid of the cancel order it get back.